src/printer.js

```diff
@@ -853,11 +863,25 @@ function printStatement(path, options, print) {
           ...(node.isFinal ? ["final"] : []),
           ...(node.isAbstract ? ["abstract"] : [])
         ];
+        const parentNode = path.getParentNode();
+        // if this is an anonymous class, we need to check if the parent was a
+        // "new" node. if it was, we need to get the arguments from that node
+        // ex: $test = new class($arg1, $arg2) extends TestClass {};
+        const anonymousArguments =
+          node.isAnonymous &&
+          parentNode.kind === "new" &&
+          parentNode.arguments.length > 0
+            ? anonymousClassesNewArguments
```

Comment:
Although now that you say that it _might_ be possible to refactor the `printArgumentsList` function to take a list of printed arguments rather than a `path`?

Reply:
Right - but basically all we'd have to do to get the state of the `path` object of the `new` node is remove the last two elements, right?

<img width="692" alt="screen shot 2018-04-06 at 19 27 38" src="https://user-images.githubusercontent.com/7962156/38435041-c1e8b290-39d0-11e8-8fa2-74a49ab1fff5.png">

Reply:
if thats true then yes that's definitely a better solution - i honestly have no clue how that stack works on `path` haha. Removing elements from it wouldn't break the class itself?

Reply:
Yeah I've also spent a lot of time trying to wrap my head around it too 😆 I could imagine that we'd have to make a copy of the path before mutating it, which makes this approach a lot less attractive.

Reply:
While we might get away without creating a deep copy and just using `slice`, we'd still have to construct a new `FastPath` object with a different stack inside. It doesn't look like that is easy (or supposed to be done, for that matter) ...

Reply:
Hmm, I didn't come up with a solution after playing around with it for a while. Let's leave it as is and hope we won't have to build a lot on top of this 😄

Reply:
👌 sounds good to me - good suggestion though! I wouldn't be opposed to digging into this down the road to see if we can make it cleaner